### PR TITLE
Handle nil language in highlighter

### PR DESCRIPTION
**Pull request by tundal45**
I noticed that `middleman build` was breaking so I thought it would be good to have that be executed as a part of the CI build. So I have added `travis.yml`. I believe there is an additional step in repository settings that needs to happen for this to work. I have also replaced the existing specs for `toc` with unit tests that do not require a local middleman server running. The hope is that this will help with https://github.com/emberjs/guides/issues/10.

### CODE REVIEW

**Comment by trek** [comment]
Some feedback on the tests: I'm not a huge fan of this level of reliance on mocks. I feel like they've described the _exact_ implementation and not the expected behavior, which makes them particularly brittle to refactoring.

Ideally the way to test this is to create a middleman site instance, add this TOC mixin, and then call methods on the instance and verify their output

**Comment by tundal45** [comment]
@trek As we discussed, the tests now primarily mock out `current_page` & `data.guides` to simulate all the edge cases for various methods in the helper. The only place `request` had to be mocked out was for `toc_for`. This should be less brittle than the earlier version. Let me know if there are still other improvements that can be made. Thanks!

**Comment by trek** [comment]
Did you forget to push some commits? I'm still only seeing the 4 commits from a few days ago.

**Comment by tundal45** [comment]
@trek I had accidentally amended the travis ci commit with the spec changes. Fixed now.

File: lib/toc.rb [hunk 1]
```diff
@@ -77,11 +67,8 @@ def guide_name
     end
 
     def chapter_name
-      if current_chapter
-        return current_chapter.title
-      else
-        return ""
-      end
+      return "" unless  current_chapter
```
Comment (trek) [inline lib/toc.rb]: We could probably nuke the returns in the original, but I don't think this adds readability

File: lib/toc.rb [hunk 2]
```diff
@@ -57,19 +57,9 @@ def toc_for(guides)
...
-          buffer << current_chapter.title
-        end
-      else
-        buffer << "Guides"
-      end
-
-      buffer
+      return "Guides" unless current_guide
+      return "#{current_guide.title}" unless current_chapter
+      "#{current_guide.title}: #{current_chapter.title}"
```
Comment (trek) [inline lib/toc.rb]: https://github.com/emberjs/guides/pull/53/files#r26567477

File: spec/toc_spec.rb [hunk 3]
```diff
@@ -1,64 +1,396 @@
...
 
-unless `which phantomjs`.empty?
-  Capybara.register_driver :poltergeist do |app|
-    Capybara::Poltergeist::Driver.new(app, { :inspector => true, })
+  before(:each) do
+    class HelperTester
+      include TOC::Helpers
+    end
+
+    allow(helper).to receive(:current_page).and_return(testing_page)
```
Comment (trek) [inline spec/toc_spec.rb]: Your comment made it sound like you got rid of this level of mocking?

File: spec/toc_spec.rb [hunk 4]
```diff
@@ -1,64 +1,396 @@
-#NOTE: start the middleman as in readme, then run bundle exec rspec
+require "spec_helper"
 
-require 'capybara/rspec'
-require 'capybara/poltergeist'
-require 'pry'
+describe Highlighter::Helpers do
```
Comment (trek) [inline spec/toc_spec.rb]: Your comment made it sound like you got rid of this level of mocking?

File: spec/toc_spec.rb [hunk 5]
```diff
@@ -1,64 +1,396 @@
...
 
-describe "TOC", :type => :feature do
-  include Capybara::DSL
+  after do
+    # Remove our testing class
+    Object.send :remove_const, :HelperTester
+  end
+
+  describe "#toc_for" do
+    let(:skip_chapter)    { double(title: "Skip Me", url: "skip-me") }
```
Comment (trek) [inline spec/toc_spec.rb]: Your comment made it sound like you got rid of this level of mocking?

File: lib/toc.rb (hunk 1 above)
Comment (tundal45) [inline lib/toc.rb]: Makes sense. I'll switch back to the old version sans explicit `return`s.

File: spec/toc_spec.rb (hunk 3 above)
Comment (tundal45) [inline spec/toc_spec.rb]: `current_page` is not something that we override in the helper and it felt was not in violation of mocking something we don't own. Is the issue that `current_page` is mocked out in a `before` block instead of each individual `example`?

Comment (trek) [inline spec/toc_spec.rb]: ah, yes, current page should stay. I guess I mostly confused by the giant `mock`/`double` block still being around?

**Comment by tundal45** [comment]
@trek I need to squash the intermediary commits but this should have tests that use actual data rather than relying heavily on mocks & stubs. It also updates the highlighter to raise an error when there are nil languages rather than silently approving them.

File: lib/highlighter.rb [hunk 6]
```diff
@@ -11,6 +12,11 @@ def registered(app)
 
   module Helpers
     def _highlight(string, language, class_name=nil)
+      error_message = "Code blocks must be fenced with proper language designated. If you don't know what language to use, specify text.\n\n"
```
Comment (trek) [inline lib/highlighter.rb]: "If you don't know what language to use, specify text" -> If you don't know what language to use, specify ```text

**Comment by tundal45** [comment]
@trek This addresses all the comments made here as well as those we discussed offline. I have [travis running builds](https://travis-ci.org/tundal45/guides/builds/55284701) on my fork as well so that should work on the main repo once this gets merged in & setup on travis.

**Comment by trek** [comment]
@tundal45 I think the ordering of PRs is wrong. We'd need to wait for a the PR that adds languages to all language-less code blocks. If we merged this first, nobody could build the site!

**Comment by trek** [comment]
Otherwise, looks good! Thanks for all the hard work on this. I know it's been a lot of back and forth.

**Comment by tundal45** [comment]
@trek Happy to help. I will open the PR that fixes the guides with language-less code blocks shortly.

**Comment by tundal45** [comment]
yayyayyay! <3
